From: Marco Felsch <m.felsch@pengutronix.de>
Date: Tue, 17 Jun 2025 13:19:11 +0200
Subject: [PATCH] drivers: tzc380: add support to dump region0

Dumping region0 is interessting too since it may have a insecure sp
configuration applied by the previous running firmware.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>
---
 core/drivers/tzc380.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/drivers/tzc380.c b/core/drivers/tzc380.c
index 44033917f73e..091fe5ad23c2 100644
--- a/core/drivers/tzc380.c
+++ b/core/drivers/tzc380.c
@@ -338,7 +338,9 @@ void tzc_dump_state(void)
 	     io_read32(tzc.base + SECURITY_INV_EN_OFF));
 	for (n = 0; n <= REGION_MAX; n++) {
 		temp_32reg = tzc_read_region_attributes(tzc.base, n);
-		if (!(temp_32reg & TZC_ATTR_REGION_EN_MASK))
+
+		/* region0 is always enabled */
+		if (n && !(temp_32reg & TZC_ATTR_REGION_EN_MASK))
 			continue;
 
 		DMSG("");
@@ -348,6 +350,9 @@ void tzc_dump_state(void)
 		DMSG("region_base: 0x%08x%08x", temp_32reg_h, temp_32reg);
 		temp_32reg = tzc_read_region_attributes(tzc.base, n);
 		DMSG("region sp: %x", temp_32reg >> TZC_ATTR_SP_SHIFT);
+		/* Skip printing the size for region0 */
+		if (!n)
+			continue;
 		DMSG("region size: %x", (temp_32reg & TZC_REGION_SIZE_MASK) >>
 				TZC_REGION_SIZE_SHIFT);
 	}
